In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
# install micromamba
!curl -Ls https://micro.mamba.pm/api/micromamba/linux-64/latest | tar -xvj bin/micromamba

bin/micromamba


In [ ]:
!./bin/micromamba create -y -n ast310 python=3.11.14


[+] 0.0s
[+] 0.1s
conda-forge/linux-64   1%
conda-forge/noarch    ⣾  [+] 0.2s
conda-forge/linux-64  15%
conda-forge/noarch    18%[+] 0.3s
conda-forge/linux-64  27%
conda-forge/noarch    42%[+] 0.4s
conda-forge/linux-64  38%
conda-forge/noarch    70%[+] 0.5s
conda-forge/linux-64  51%
conda-forge/noarch    95%conda-forge/noarch                                
[+] 0.6s
conda-forge/linux-64  54%[+] 0.7s
conda-forge/linux-64  77%[+] 0.8s
conda-forge/linux-64  99%[+] 0.9s
conda-forge/linux-64  99%conda-forge/linux-64                              


Transaction

  Prefix: /root/.local/share/mamba/envs/ast310

  Updating specs:

   - python=3.11.14


  Package              Version  Build                 Channel          Size
─────────────────────────────────────────────────────────────────────────────
  Install:
─────────────────────────────────────────────────────────────────────────────

  + _libgcc_mutex          0.1  conda_forge           conda-forge       3kB
  + _openmp_mutex          4.

In [11]:
!export PATH=/content/bin:$PATH && micromamba run -n ast310 python --version

Python 3.11.14


In [ ]:
%ls

bin/  drive/  sample_data/


In [ ]:
!export PATH=/content/bin:$PATH && micromamba run -n ast310 pip install mutagen

In [ ]:
!export PATH=/content/bin:$PATH && micromamba run -n ast310 pip install -r /content/drive/MyDrive/GraFP/prova.txt


  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 18.3/18.3 MB 155.7 MB/s  0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 38.4/38.4 MB 148.0 MB/s  0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 23.8/23.8 MB 106.2 MB/s  0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 6.6/6.6 MB 116.0 MB/s  0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.3/3.3 MB 134.8 MB/s  0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 534.5/534.5 kB 27.4 MB/s  0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 56.3/56.3 MB 119.9 MB/s  0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 536.2/536.2 kB 27.6 MB/s  0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.1/2.1 MB 98.9 MB/s  0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.7/3.7 MB 92.3 MB/s  0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 7.0/7.0 MB 186.9 MB/s  0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━

In [ ]:
%cd /content/drive/MyDrive/GraFP

/content/drive/.shortcut-targets-by-id/1suc2HZBEmgYV_AfH3YEB6lDfyHuc-ofZ/GraFP


In [ ]:
!export PATH=/content/bin:$PATH && micromamba run -n ast310 python /content/drive/MyDrive/GraFP/convert_to_mm.py

Saved: output/db.mm
Saved: output/db_shape.npy (shape=(54426, 128), dtype=float32)


In [ ]:
!export PATH=/content/bin:$PATH && micromamba run -n ast310 python /content/drive/MyDrive/GraFP/generate.py --test_dir /content/drive/MyDrive/GraFP/fma_small

Loading model...
path is /content/drive/MyDrive/GraFP/fma_small
=>Loading indices from /content/drive/MyDrive/GraFP/fma_small
len before: 8008
len after: 8008
Loaded 8008 files from /content/drive/MyDrive/GraFP/fma_small
/root/.local/share/mamba/envs/ast310/lib/python3.11/site-packages/torch_audiomentations/core/composition.py:42: FutureWarning: Transforms now expect an `output_type` argument that currently defaults to 'tensor', will default to 'dict' in v0.12, and will be removed in v0.13. Make sure to update your code to something like:
  >>> augment = Compose(..., output_type='dict')
  >>> augmented_samples = augment(samples).samples
  warnings.warn(
Loading checkpoint...
=> loading checkpoint '/content/drive/MyDrive/GraFP/checkpoint/model_tc_27_best.pth'
Computing fingerprints...
Step [0/8008]	 shape: torch.Size([47, 128])
Step [10/8008]	 shape: torch.Size([46, 128])
Step [20/8008]	 shape: torch.Size([46, 128])
Step [30/8008]	 shape: torch.Size([46, 128])
Step [40/8008]	 shape: tor

In [ ]:
!export PATH=/content/bin:$PATH && micromamba run -n ast310 python /content/drive/MyDrive/GraFP/test_fp.py --test_dir fma_mini --n_query_db 80

In [ ]:
!export PATH=/content/bin:$PATH && micromamba run -n ast310 python /content/drive/MyDrive/GraFP/query.py --test_ids all --test_dir fma_mini/002

Creating new model...
Creating dataloaders ...
path is fma_mini/002
=>Loading indices from fma_mini/002
Loaded 3 files from fma_mini/002
=> loading checkpoint '/content/drive/MyDrive/GraFP/checkpoint/model_tc_27_best.pth'
=> Creating query and db fingerprints...
Step [0/3]	 shape: torch.Size([302, 128])
Step [1/3]	 shape: torch.Size([302, 128])
Step [2/3]	 shape: torch.Size([303, 128])
Load 907 items from inference/1768994462/query.mm.
Load 907 metadata items from inference/1768994462/metadata.npy.
Load 1,210 items from /content/drive/MyDrive/GraFP/output/db.mm.
Load 1,210 metadata items from /content/drive/MyDrive/GraFP/output/db_metadata.npy.
[['....like bacon sundays']
 ['....like bacon sundays']
 ['....like bacon sundays']
 ...
 ['my sweet miasma']
 ['my sweet miasma']
 ['my sweet miasma']]
Creating index: ivfpq
Training index...
WARNING clustering 1210 points to 64 centroids: please provide at least 2496 training points
WARNING clustering 1210 points to 256 centroids: please provi

Load 54,426 items from logs/store/small/model_tc_27_best/query.mm.
Load 54,426 metadata items from logs/store/small/model_tc_27_best/metadata.npy.
Traceback (most recent call last):
  File "/content/drive/MyDrive/GraFP/query.py", line 244, in <module>
    main()
  File "/content/drive/MyDrive/GraFP/query.py", line 236, in main
    predict_query(emb_dir=fp_dir,
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/content/drive/MyDrive/GraFP/query.py", line 64, in predict_query
    dummy_db, dummy_db_shape, dummy_db_metadata = load_memmap_data(emb_dummy_dir, 'dummy_db')
                                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/content/drive/.shortcut-targets-by-id/1suc2HZBEmgYV_AfH3YEB6lDfyHuc-ofZ/GraFP/eval.py", line 166, in load_memmap_data
    path_shape = os.path.join(source_dir, fname + '_shape.npy')
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen posixpath>", line 76, in join
TypeError: expected str, bytes or os.PathLike object, not NoneType

In [ ]:
import numpy as np

arr = np.load("/content/drive/MyDrive/GraFP/logs/store/small/model_tc_27_best/5b5ad1aa/hit_rates.npy")
print(arr)


[[72.65 81.1  84.05 86.7  87.7  91.25]
 [90.25 94.3  96.   96.9  97.5  98.65]
 [91.55 94.95 96.45 97.45 98.1  99.1 ]
 [94.65 97.15 97.95 98.7  99.15 99.9 ]]


In [ ]:
arr = np.load("/content/drive/MyDrive/GraFP/logs/store/small/model_tc_27_best/0b2093c6/gt_ids.npy")
print(arr[:20])

[36372 36311 51428 39769 39102 51783 53302 39274 46108 34492 51326 46166
 45636 39664 35183 51712 46855 52407 39510 32155]


In [ ]:
arr = np.load("/content/drive/MyDrive/GraFP/logs/store/small/model_tc_27_best/0b2093c6/pred_ids.npy")
print(arr[:20])

[[36371 36371 36371 36371 36371 36371]
 [36311 36311 36311 36311 36311 36311]
 [51429 51429 51428 51428 51428 51428]
 [39768 39768 39768 39769 39769 39769]
 [39102 39102 39102 39102 39102 39102]
 [51784 51783 51783 51783 51783 51783]
 [53302 53302 53302 53302 53302 53302]
 [39274 39274 39274 39274 39274 39274]
 [46108 46108 46108 46108 46108 46108]
 [34491 34492 34492 34492 34492 34492]
 [51325 51325 51325 51325 51325 51326]
 [46166 46166 46166 46166 46166 46166]
 [45635 45635 45635 45635 45635 45635]
 [39664 39664 39664 39664 39664 39664]
 [35183 35182 35182 35183 35182 35182]
 [51712 51712 51712 51712 51712 51712]
 [46855 46855 46855 46855 46855 46855]
 [52407 52407 52407 52407 52407 52407]
 [39510 39510 39510 39510 39510 39510]
 [32155 32155 32155 32155 32155 32155]]


In [ ]:
arr = np.load("/content/drive/MyDrive/GraFP/output/db_metadata.npy")
print(arr.shape)
print(arr[:20])

(1210, 1)
[['....like bacon sundays']
 ['....like bacon sundays']
 ['....like bacon sundays']
 ['....like bacon sundays']
 ['....like bacon sundays']
 ['....like bacon sundays']
 ['....like bacon sundays']
 ['....like bacon sundays']
 ['....like bacon sundays']
 ['....like bacon sundays']
 ['....like bacon sundays']
 ['....like bacon sundays']
 ['....like bacon sundays']
 ['....like bacon sundays']
 ['....like bacon sundays']
 ['....like bacon sundays']
 ['....like bacon sundays']
 ['....like bacon sundays']
 ['....like bacon sundays']
 ['....like bacon sundays']]
